File: 1299. Replace Elements with Greatest Element on Right Side.ts
function replaceElements(arr: number[]): number[] {
    const res:number[] = new Array(arr.length);
    res[arr.length - 1] = -1;
    for (let i:number = arr.length - 1; i > 0; i --) {
    res[i - 1] = Math.max(arr[i], res[i]);
    }
    return res;    
};

File: 1299. Replace Elements with Greatest Element on Right Side.kt
class Solution {
    fun replaceElements(arr: IntArray): IntArray {
        val res = IntArray(arr.size)
        res[arr.size - 1] = -1
        for (i in arr.size - 1 downTo 1) {
            res[i - 1] = maxOf(arr[i], res[i])
        }
        return res 
    }
}
